docs/ExtendingHeartbeatProperties.md

```diff
@@ -0,0 +1,96 @@
+status of those properties for the duration of the application life cycle.
+
+To add the payload properties, aquire the IHeartbeatPropertyManager module using the internal
+`TelemetryModules` singleton. Do this in your `Initialize(TelemetryConfiguration telemetryConfig)` implementation method, and iterate through the available modules looking for an implementation of `IHeartbeatPropertyManager`:
+
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    ...
+    // heartbeat property manager to add/update my health properties into:
+    private IHeartbeatPropertyManager heartbeatManager= null;
+    ...
+    public void Initialize(TelemetryConfiguration configuration)
+    {
+        ...
+        var modules = TelemetryModules.Instance;
+        foreach (var telemetryModule in modules.Modules)
+        {
+            if (telemetryModule is IHeartbeatPropertyManager)
+            {
+                this.heartbeatManager = telemetryModule as IHeartbeatPropertyManager;
+                break;
+            }
+        }
+        ...
+
+...now you will have a heartbeat property manager that you can work with (or not, remember to
+test!). From here (but likely still within your `Initialize` method, you can Add the health
+property fields that you want to see in the heartbeat payload for the duration of the
+application's lifecycle.
+
+    ...
+    this.heartbeatManager.AddHealthProperty(propertyName: "myHealthProperty", propertyValue: this.MyHealthProperty, isHealthy: true);
+    ...
+
+Outside of your `Initialize` method, you can update the values you've added by using the `SetHealthProperty` method very simply. For instance, you can add a health property called
+'myHealthProperty' in the initialize method as above, and then from within a property elsewhere
+in your class, you can update the value in the heartbeat payload as follows:
+
+    public string MyHealthProperty
+    {
+        get => this.myHealthPropertyValue;
+
+        set
+        {
+            this.myHealthPropertyValue = value;
+            if (this.heartbeatManager != null)
+            {
+                bool myPropIsHealthy = this.SomeTestForHealthStatus(this.myHealthPropertyValue);
+                this.heartbeatManager.SetHealthProperty(propertyName: "myHealthProperty", propertyValue: value, isHealthy: myPropIsHealthy);
+            }
+        }
+    }
+
+Using the above example you can add and update properties in the Health Heartbeat for the
+duration of your application's life.
+
+> **Note:** You may also set values for the `HeartbeatInterval` value. This is discouraged, as
+your override of this value may adversely affect the consumer's ApplicationInsights.config
+configuration in doing so.
+
+You can also set values into the `ExcludedDefaultProperties` list if you find it pertinent to
+do so.  Setting values into the
+`ExcludedDefaultProperties` is fine, as your module may provide more detailed information about
```

Comment:
Did you mean ExcludedHeartbeatProperties?

Reply:
Yes, thanks!